portfolio/src/main/webapp/script.js

```diff
@@ -199,16 +244,69 @@ async function deleteMessage(msg) {
   });
 }
 
-function openReplies(msgID, tag) { 
-  document.getElementById("blogcontent").style.display = "none";
-  document.getElementById("blogreplycontent").style.display = "block";
-  document.getElementById('parentID').value = msgID;
-  document.getElementById('tags-reply').value = tag;
+// Displays the reply form under a post when its reply button is clicked.
+// Removes the reply form if button is clicked again, and so on...
+function handleReplies(msgID, msgSender, tag, postEleID) { 
+  // The form appears when a reply button is clicked an even number of times, so
+  // If a users clicks a different reply button, set counter to zero for that button
+  if (clickedDifferentReplyButton(msgID)) {
+    replyClickCounter = 0;
+  }
+  
+  // Remove the form when a button is clicked an odd number of times in a row.
+  if (replyClickCounter % 2 != 0) {
+    document.getElementById('reply-form').style.display = 'none';
+    return
+  }
+  
+  // Put reply form under the post.
+  let replyNode = document.getElementById('reply-form');
+  var referenceNode = document.querySelector(`#${CSS.escape(postEleID)}`);
+  referenceNode.parentNode.insertBefore(replyNode, referenceNode.nextSibling);
+  replyNode.style.display = 'block';
+  
+  // Show submit button only if a user types, 
```

Comment:
I think you can 'grey' out the buttons somehow. It might be possible to make the text box a required field which would do the same check as you have below.

Reply:
Oh yeah grey-ing the button out would be great